Doc/config-api.tex

```diff
@@ -604,87 +235,46 @@ \subsection{Configuration Option Types}
 value that has a way to turn debugging information on (\longprogramopt{debug}) 
 or off (\longprogramopt{no-debug}).
 \begin{verbatim}
-BooleanOption( options = '--debug !--no-debug' )
+BooleanOption("", options = '--debug !--no-debug', default=True)
 \end{verbatim}
 \end{classdesc}
 
-\begin{classdesc}{CompoundOption}{\optional{\class{GenericOption} arguments}}
-Compound options are options that contain multiple elements on the 
-command-line.  They are simply groups of command-line arguments surrounded
-by a pair of grouping characters (e.g. (~), [~], \{~\}, <~>).  This grouping
-can contain anything including other command-line arguments.  However, 
-all content between the grouping characters is unparsed.  This can be useful
-if you have a program that wraps another program and you want to be able
-to forward the wrapped program's options on.  An example of a compound option
-used on the command-line is shown below.
-\begin{verbatim}
-# Capture the --diff-opts options to send to another program
-mycommand --other-opt --diff-opts ( -ib --minimal ) file1 file2
-\end{verbatim}
-\end{classdesc}
+\begin{classdesc}{MultiStringOption}{\optional{\class{GenericOption} arguments}}
+A \class{MultiStringOption} is an option that is intended to be used multiple
+times on the command-line, or take a list of values. Other options when
+specified more than once simply overwrite the previous value.
+\class{MultiStringOption}s will append the new values to a list.
 
-\begin{classdesc}{CountedOption}{\optional{\class{GenericOption} arguments}}
-A \class{CountedOption} is a boolean option that keeps track of how many
-times it has been specified.  This is useful for options that control 
-the verbosity of logging messages in a program where the number of times
-an option is specified, the more logging information is printed.
-\end{classdesc}
+In the configuration file, this is a space separated list, interpreted in the
+same ways as shell arguments (with the same escaping/quoting rules for strings
+that contains whitespaces). In fact, it is processed by \code{shlex.split}.
 
-\begin{classdesc}{InputDirectoryOption}{\optional{\class{GenericOption} arguments}}
-An \class{InputDirectoryOption} is an option that accepts a directory name
-for input.
-This directory name is checked to make sure that it exists and that it is
-readable.  If it is not, a \exception{InvalidOptionError} exception is 
-raised.
+The individual values of \class{MultiStringOption} are strings. This can, of
+course, be made generic, but such a use case has not arisen.
 \end{classdesc}
 
-\begin{classdesc}{OutputDirectoryOption}{\optional{\class{GenericOption} arguments}}
-An \class{OutputDirectoryOption} is an option that accepts a directory name
-for output.  If the directory exists, it is checked to make sure that it is
-readable.  If it does not exist, it is created.
+\begin{classdesc}{DictOption[T]}{\optional{\class{GenericOption} arguments}}
```

Comment:
CompoundOption became DictOption. It is up to individual commands to decide how this is implemented (e.g. links takes in 2-3 arguments in each group while counter takes in 2). You can see the documentation for the links and counter commands for examples.

It is intended to be used with a section with exactly one option, but can in practice be used when there are other options. In a config file, options in the section that do not refer to an existent option are put in the first DictOption. I figured it would be easier to add a `DictOption` than subclassing `ConfigSection` further

Reply:
I think this is fair as long as it's clearly documented. I know this section is about the ConfigManager *API*, but I think you can still explain the relevant part of the default config file, if only as an example.